app/src/main/java/org/fossasia/openevent/general/about/AboutEventActivity.kt

```diff
@@ -58,11 +58,25 @@ class AboutEventActivity : AppCompatActivity(), AppBarLayout.OnOffsetChangedList
         val dateString = StringBuilder()
         val startsAt = EventUtils.getLocalizedDateTime(event.startsAt)
         val endsAt = EventUtils.getLocalizedDateTime(event.endsAt)
-        aboutEventDetails.text = dateString.append(EventUtils.getFormattedDate(startsAt))
-                                           .append(" - ")
-                                           .append(EventUtils.getFormattedDate(endsAt))
-                                           .append(" • ")
-                                           .append(EventUtils.getFormattedTime(startsAt))
+        if (EventUtils.getFormattedDate(startsAt) != EventUtils.getFormattedDate(endsAt)) {
+            aboutEventDetails.text = dateString.append(EventUtils.getFormattedDate(startsAt))
+                    .append(" at ")
+                    .append(EventUtils.getFormattedTime(startsAt))
+                    .append(" - ")
+                    .append(EventUtils.getFormattedDate(endsAt))
+                    .append(" at ")
+                    .append(EventUtils.getFormattedTime(endsAt))
+                    .append(" ")
+                    .append(EventUtils.getFormattedTimeZoneWithBrackets(endsAt))
+        } else {
+            aboutEventDetails.text = dateString.append(EventUtils.getFormattedDate(startsAt))
+                    .append(" from ")
+                    .append(EventUtils.getFormattedTime(startsAt))
+                    .append(" to ")
+                    .append(EventUtils.getFormattedTime(endsAt))
+                    .append(" ")
+                    .append(EventUtils.getFormattedTimeZoneWithBrackets(endsAt))
```

Comment:
I mean it should be `EventUtils.getFormattedEventDateRange` and it should accept the `startsAt` and `endsAt` variables and then if the startsAt and endsAt dates are same then return the String according to the `if` part otherwise the `else` part. 
The code in activity will just be `EventUtils.getFormattedEventDateRange(startsAt, endsAt)`. All the `if`-`else` will be handled inside the Utils file only.